crates/weaver_live_check/src/advice.rs

```diff
@@ -68,33 +69,54 @@ impl Advisor for DeprecatedAdvisor {
     fn advise(
         &mut self,
         sample: SampleRef<'_>,
+        signal: &Sample,
         registry_attribute: Option<Rc<Attribute>>,
         registry_group: Option<Rc<ResolvedGroup>>,
     ) -> Result<Vec<Advice>, Error> {
         match sample {
-            SampleRef::Attribute(_sample_attribute) => {
+            SampleRef::Attribute(sample_attribute) => {
                 let mut advices = Vec::new();
                 if let Some(attribute) = registry_attribute {
                     if let Some(deprecated) = &attribute.deprecated {
                         advices.push(Advice {
                             advice_type: "deprecated".to_owned(),
-                            value: deprecated_to_value(deprecated),
-                            message: deprecated.to_string(),
+                            value: json!({
+                                "attribute_name": sample_attribute.name.clone(),
+                                "deprecation_reason": deprecated_to_reason(deprecated),
+                                "deprecation_note": deprecated.to_string(),
+                            }),
+                            message: format!(
+                                "Attribute '{}' is deprecated; reason = '{}', note = '{}'.",
+                                sample_attribute.name.clone(),
+                                deprecated_to_reason(deprecated),
+                                deprecated
+                            ),
                             advice_level: AdviceLevel::Violation,
+                            signal_type: signal.signal_type(),
+                            signal_name: signal.signal_name(),
                         });
                     }
                 }
                 Ok(advices)
             }
-            SampleRef::Metric(_sample_metric) => {
+            SampleRef::Metric(sample_metric) => {
                 let mut advices = Vec::new();
                 if let Some(group) = registry_group {
                     if let Some(deprecated) = &group.deprecated {
                         advices.push(Advice {
                             advice_type: "deprecated".to_owned(),
-                            value: deprecated_to_value(deprecated),
-                            message: deprecated.to_string(),
+                            value: json!({
+                                "deprecation_reason": deprecated_to_reason(deprecated),
+                                "deprecation_note": deprecated,
+                            }),
+                            message: format!(
+                                "Metric is deprecated; reason = {}, note = {}",
```

Comment:
For Attributes we're repeating the name in the message and context: `"Attribute '{}' is deprecated; reason = '{}', note = '{}'.",` should we do the same for Metric names, or remove the repeat for Attribute?